cart/interfaces/controller/cartcontroller.go

```diff
@@ -211,3 +236,25 @@ func (cc *CartViewController) CleanDeliveryAndViewAction(ctx context.Context, r
 
 	return cc.responder.RouteRedirect("cart.view", nil)
 }
+
+// AdjustItemsToRestrictedQtyAndAddToSession checks the items of the cart against their qty restrictions and adds adjustments to the session
+func (cc *CartViewController) AdjustItemsToRestrictedQtyAndAddToSession(ctx context.Context, r *web.Request) (application.AdjustmentResults, error) {
+	adjustments, err := cc.applicationCartService.AdjustItemsToRestrictedQty(ctx, r.Session())
+	if err != nil {
+		return adjustments, err
+	}
+
+	cc.addAdjustmentsToSession(adjustments, r)
+
+	return adjustments, nil
+}
+
+func (cc *CartViewController) addAdjustmentsToSession(adjustments application.AdjustmentResults, r *web.Request) {
+	for _, a := range adjustments {
+		if a.WasDeleted {
+			r.Session().AddFlash(a, "cart.view.adjustment.delete")
+		}
+	}
+
+	r.Session().Store("cart.view.adjustment.update", adjustments)
+}
```

Comment:
I agree that the update could also be served via flash, as the client does not not otherwise when this has changed. Maybe we can discuss the requirements behind this again?

Reply:
Othwise this needs to be cleaned up regularly

Reply:
Provided 3 template functions that enable the frontend to manage (read and delete) the adjustment messages.